web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -71,6 +75,20 @@ public GraphHopperManaged(CmdArgs configuration, ObjectMapper objectMapper) {
                 throw new RuntimeException(e);
             }
         }
+
+        String flexModelLocation = configuration.get("graph.flex_models.directory", "");
```

Comment:
Why do we have to put a model in the flex_models directory ? Is the idea to first put a preset flex model in the dir and then customize this even more per request ? I would have expected to be able to just put something like `base: mtb` on the request without having pre-configured anything on the server side (empty flex_model directory). I see a preconfigured flexmodel provided with the server is also useful, but should it not work without it as well (using the standard GH vehicles as possible base) ?

Btw could you add some docs to config-example.yml already ? This will help to get started more quickly for everybody trying the new feature (including me of course ;)). We should also improve the error messages. The first thing I tried was simply typing `base: mtb` into the new text box in the browser, but this neither worked nor did I get a message what I was doing wrong. Its a bit confusing at the moment having the url parameters, the text box and the flex_models directory (not talking about the ch profiles even).

Reply:
> Why do we have to put a model in the flex_models directory ? Is the idea to first put a preset flex model in the dir and then customize this even more per request ?

No. This is completely independent. So if you do not use flex_models directory you can still do flex requests. BUT now let's assume you are fine with the result and want to create a CH or LM version on it then you put the request into a file in this flex_models directory. Of course we can change naming here if this is confusing.

> This will help to get started more quickly for everybody trying the new feature (including me of course ;))

Oh, sure. I totally neglected this part.

> Its a bit confusing at the moment having the url parameters, the text box 

Yeah, I understand this. This is also more a way to make it done very quickly and we should probably think of a different way. Only the text box stuff and the points from the URL are used: https://github.com/graphhopper/graphhopper/pull/1841/files#diff-9941521f8b7f449ab4c126d163e4d94aR129

Reply:
> Why do we have to put a model in the flex_models directory ? 

Sorry I missed that part. I just got flexModel==null and thought it has to be in the directory. But if there is a flexModel set on the request it is always used.

> BUT now let's assume you are fine with the result and want to create a CH or LM version on it then you put the request into a file in this flex_models directory. Of course we can change naming here if this is confusing.

For CH and LM we need to put the flex model somewhere for the preparation yes. So far I thought we put it into config.yml, but ok putting it into a separate folder and referencing this in the config.yml is basically the same. When the server starts / during import these flex models are evaluated and considered for the import (and maybe at some point we will have a POST endpoint that allows sending a flex model and preparing CH/LM for it). 
But why do we look at the flex directory again at request time ?

Reply:
> So far I thought we put it into config.yml

It was just easier to do it in a separate folder and now I also think this might be better. But as everything here it is not set in stone :)

> But why do we look at the flex directory again at request time ?

Oh, do we :) ?

Reply:
> It was just easier to do it in a separate folder and now I also think this might be better.

Yes I think that's fine

> Oh, do we :) ?

Yes, in `GraphHopper#createWeighting`

Reply:
> > Oh, do we :) ?

> Yes, in GraphHopper#createWeighting

We do not parse yaml from disk for a request. We only use previously defined FlexModels. E.g. if you have defined truck.yml then a GET request with `vehicle=truck` should behave like "POST /route with the content of truck.yml" (and also a POST request with `base: truck` should load previously defined values).